projects/ui-policy-studio-angular/src/lib/components/flows-menu/gio-ps-flows-menu.component.ts

```diff
@@ -188,6 +211,56 @@ export class GioPolicyStudioFlowsMenuComponent implements OnChanges {
         return true;
       };
     }
+    this.applyFilter(this.searchControl.value || '');
+  }
+
+  private applyFilter(searchTerm: string): void {
+    const term = searchTerm.toLowerCase();
+
+    if (!term) {
+      this.filteredFlowGroupMenuItems = [...this.flowGroupMenuItems];
+      return;
+    }
+
+    this.filteredFlowGroupMenuItems = this.flowGroupMenuItems
+      .map(group => {
+        const filteredFlows = group?.flows?.filter(flow => {
+          if (group.name && group.name?.toLowerCase().includes(term)) return true;
+          if (flow.name && flow.name?.toLowerCase().includes(term)) return true;
+          if (flow.pathOrChannelLabel && flow.pathOrChannelLabel?.toLowerCase().includes(term)) return true;
+          if (flow.selectors && flow.selectors?.some(selector => this.selectorMatches(selector, term, flow.hasCondition))) return true;
+          if (flow.badges && flow.badges?.some(badges => badges.label?.toLowerCase().includes(term))) return true;
+          return false;
+        });
+        return { ...group, flows: filteredFlows };
+      })
+      .filter(group => group.flows.length > 0);
+  }
+
+  private selectorMatches(selector: Selector, term: string, hasCondition: boolean): boolean {
+    const type = selector.type;
+
+    if (type === 'HTTP') {
+      const httpSelector = selector as HttpSelector;
+      if (httpSelector.path?.toLowerCase().includes(term)) return true;
+      if (httpSelector.methods?.some(method => method.toLowerCase().includes(term))) {
+        return true;
+      }
+    }
+
+    if (type === 'CHANNEL') {
+      const channelSelector = selector as ChannelSelector;
+      if (channelSelector.channel?.toLowerCase().includes(term)) return true;
+      if (channelSelector.operations?.some(op => op.toLowerCase().includes(term))) {
+        return true;
+      }
+    }
+
+    if (hasCondition && type === 'CONDITION') {
+      const conditionSelector = selector as ConditionSelector;
+      if (conditionSelector.condition?.toLowerCase().includes(term)) return true;
```

Comment:
No need to research this, it doesn't make sense and may bring up more results than necessary imho